src/gui/tray/ColorSvgImageProvider.h

```diff
@@ -1,8 +1,11 @@
 #pragma once
 
-#include <qimage.h>
-#include <qpalette.h>
-#include <qquickimageprovider.h>
+#include <QImage>
+#include <QPalette>
+#include <QQuickImageProvider>
+#include <QQuickAsyncImageProvider>
+#include <QThreadPool>
```

Comment:
You overdid it here I think. You don't need the QThreadPool. Most of your issues are likely coming from the multiple inheritance between QRunnable and QObject. This is generally a bad idea, and if you *really* need to do it (not your case here) this requires extra care to make sure slots are processed on the proper thread.